1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,35 @@
-function setAlarm() {}
+let backgroundFlashing;
+function changeBackgroundColor() {
+ backgroundFlashing = setInterval(() => {
+    var randomColor = Math.floor(Math.random() * 16777215).toString(16);
+     document.body.style.backgroundColor = `#${randomColor}`;
+  },1000);
+}
+function stopFlashing() {
+  clearInterval(backgroundFlashing);
+  document.body.style.backgroundColor = 'white';
+}
+function setAlarm() {
+ 
+  let btnSet = document.getElementById('set');
+  let h1TimeRemaining = document.getElementById('timeRemaining');
+  let inputAlarmSet = document.getElementById('alarmSet');
+  let date = new Date("July 21, 1983 00:00:00");
+    date.setSeconds(inputAlarmSet.value);
+    h1TimeRemaining.innerHTML = 'Time Remaining: ' + date.toString().substring(19, 24);
+   let myInterval = setInterval(() => {
+      date.setSeconds(date.getSeconds() - 1);
+      h1TimeRemaining.innerHTML =
+        "Time Remaining: " + date.toString().substring(19, 24);
+      if (date.toString().substring(19, 24) === "00:00") {
+        playAlarm();
+        changeBackgroundColor();
+        clearInterval(myInterval);
+      }
+    }, 1000)
+    
+   
+}
```

Comment:
Hi @AmarachiOkereke, well done on getting this to work!  What I was getting at on the homework call is that it is not necessary to use a `Date` object here, and in fact the hard coded arbitrary date of `"July 21, 1983 00:00:00"` might be a bit confusing to future developers.  The `setInterval(callback, milliseconds)` function already comes with a built-in concept of time via it's second argument of milliseconds (`1000` so 1 second in this case).  We can use this to decrement a variable that we set without having to use Date objects.  I think if we reworked your `setAlarm()` function to look like this, it would still work but we'd remove some complexity around Date objects:

```
function setAlarm() {
  let h1TimeRemaining = document.getElementById("timeRemaining");
  let inputAlarmSet = document.getElementById("alarmSet");
  let seconds = inputAlarmSet.value;
  h1TimeRemaining.innerHTML = "Time Remaining: " + seconds;
  let myInterval = setInterval(() => {
    seconds -= 1;
    h1TimeRemaining.innerHTML = "Time Remaining: " + seconds;
    if (seconds === 0) {
      playAlarm();
      changeBackgroundColor();
      clearInterval(myInterval);
    }
  }, 1000);
}
```
Let me know if that makes sense!